src/axom/core/ArrayView.hpp

```diff
@@ -54,10 +55,20 @@ class ArrayView : public ArrayBase<T, DIM, ArrayView<T, DIM>>
   template <typename... Args>
   ArrayView(T* data, Args... args);
 
+  /*! 
+   * \brief Constructor for transferring between memory spaces
+   * 
+   * \param [in] other The array in a different memory space to copy from
+   * 
+   * \note The parameter is non-const because \a other can be modified through the constructed View
+   */
+  template <typename OtherArrayType>
+  ArrayView(ArrayBase<T, DIM, OtherArrayType>& other);
```

Comment:
My understanding is that this constructor is intentionally not marked as `explicit`.

In light of #693, please add a comment on this and any other constructors for the Array classes that are intentionally implicit.